pyamg/aggregation/smooth.py

```diff
@@ -430,35 +476,40 @@ def cg_prolongation_smoothing(A, T, B, BtBinv, Sparsity_Pattern, maxiter, tol,
                                                int(T.shape[1]/T.blocksize[1]),
                                                A.blocksize[0], A.blocksize[1],
                                                P.blocksize[1])
+        cost[0] += mat_mat_complexity(A,AP,incomplete=True) / float(A.nnz)
 
         # Enforce AP*B = 0
         Satisfy_Constraints(AP, B, BtBinv)
+        cost[0] += (AP.nnz * (2.0*B.shape[1] + B.shape[1]**2) + \
+                    (B.shape[1]**3) * B.shape[0] ) / float(A.nnz)
+
 
         # Frobenius inner-product of (P, AP)
         alpha = newsum/(P.conjugate().multiply(AP)).sum()
+        cost[0] += max(P.nnz, AP.nnz) / float(A.nnz)
 
         # Update the prolongator, T
         T = T + alpha*P
+        cost[0] += max(P.nnz, T.nnz) / float(A.nnz)
 
         # Ensure identity at C-pts
         if Cpt_params[0]:
             T = Cpt_params[1]['I_F']*T + Cpt_params[1]['P_I']
+            cost[0] += T.nnz / float(A.nnz)
```

Comment:
I would probably not count this as a "cost", in that for a real code, you'd modify the above 
T + alpha*P 
operation so that it doesn't disturb the identity at C-points. This is just done for convenience.